cv_next/app/feed/components/filterPanel.tsx

```diff
@@ -0,0 +1,46 @@
+import { useEffect, useState } from "react"
+import { SearchBox } from "../../../components/ui/filters/searchbar"
+import { DropdownInput } from "../../../components/ui/filters/valueSelect"
+import Categories from "@/types/models/categories"
+import { ModeSwitch } from "@/components/ui/filters/modeSwitch"
+
+export interface filterObj {
+    searchValue: string
+    categoryId: number | null
+}
+
+export const MODES:[string, string] = ["review", "published"]
+
+export const FilterPanel = ({defaultFilters, onChange, cvs}: {
+    defaultFilters: filterObj
+    cvs: CvModel[],
+    onChange: (filters: filterObj) => void
+  }) => {
+    const [searchValue, setSearchValue] = useState(defaultFilters.searchValue)
+    const [categoryId, setCategoryId] = useState(defaultFilters.categoryId)
+    const [modeValue, setModeValue] = useState(MODES[0])
+
+    useEffect(() => {
+        onChange({
+            categoryId: categoryId,
+            searchValue: searchValue
+        })
+    }, [searchValue, categoryId])
+
+    return (
+        <div className="mx-10 my-[10px] flex flex-row justify-between items-center gap-2">
+            <SearchBox placeHolder=" input text to search" value={searchValue} onChange={(value: string) => {setSearchValue(value)}}></SearchBox>
```

Comment:
Tip: for all the `onChange`s, you're probably fine with putting the setState function itself (e.g. `onChange={setSearchValue}`)

Reply:
Also can this be stored in the component logic and not the render code? `{Object.keys(Categories.category).map(key=> parseInt(key)).filter(key => !isNaN(key))}`
Might wanna memoize it at some point, so it'd be useful to keep it separated anyway (originally this is a code-style suggestion, but it's also a utilization suggestion).

Reply:
Updated the onchange code and added memoization for the filtering(which is now also separated)